src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,256 @@
+
+            string modifiedNugetConfig = Path.Combine(PackageSourceMappingsSetup.PackageSourceMappingsRoot, nugetConfigFilename);
+            Directory.CreateDirectory(Path.GetDirectoryName(modifiedNugetConfig)!);
+            File.Copy(originalNugetConfig, modifiedNugetConfig, true);
+            UpdateNugetConfigTokens(modifiedNugetConfig);
+
+            var task = new UpdateNuGetConfigPackageSourcesMappings()
+            {
+                SbrpCacheSourceName = "source-build-reference-package-cache",
+                SbrpRepoSrcPath = TestSetup.SourceBuildReferencePackagesRepo,
+                SourceBuiltSourceNamePrefix = "source-built-",
+                NuGetConfigFile = modifiedNugetConfig,
+                BuildWithOnlineFeeds = useOnlineFeeds,
+                SourceBuildSources = sources,
+                CustomSources = customSources
+            };
+
+            if (sourceBuild)
+            {
+                task.ReferencePackagesSourceName = ReferencePackagesSourceName;
+                task.PreviouslySourceBuiltSourceName = PreviouslySourceBuiltSourceName;
+                task.PrebuiltSourceName = PrebuiltSourceName;
+            }
+
+            task.Execute();
+
+            TokenizeLocalNugetConfigFeeds(modifiedNugetConfig);
+
+            string expectedNugetConfigContents = File.ReadAllText(expectedNugetConfig);
+            string modifiedNugetConfigContents = File.ReadAllText(modifiedNugetConfig);
+            Assert.Equal(expectedNugetConfigContents, modifiedNugetConfigContents);
+        }
+
+        private static void UpdateNugetConfigTokens(string nugetConfigFile)
+        {
+            ApplyLocalTokenSourceMappings(nugetConfigFile, updateTokens: true);
+        }
+
+        private static void TokenizeLocalNugetConfigFeeds(string nugetConfigFile)
+        {
+            ApplyLocalTokenSourceMappings(nugetConfigFile, tokenize: true);
+        }
+
+        private static void ApplyLocalTokenSourceMappings(string nugetConfigFile, bool updateTokens = false, bool tokenize = false)
+        {
+            if (updateTokens == tokenize)
+            {
+                throw new InvalidOperationException($"One and only one option should be true, '{nameof(updateTokens)}' or '{nameof(tokenize)}'");
+            }
+
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = updateTokens
+                    ? fileContents.Replace(kvp.Key, kvp.Value)
+                    : fileContents.Replace(kvp.Value, kvp.Key);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        internal class PackageSourceMappingsSetup
+        {
+            private static PackageSourceMappingsSetup? instance;
+            private static readonly object myLock = new();
+            private Dictionary<string, string>? localTokenSourceMappings;
```

Comment:
I find it better to separate the static versus instance members as it helps the readability.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/f51cb511fdcb7b01d02ba16d8f880d7793248d09